### PR TITLE
fix: go enum schema changes

### PR DESCRIPTION
**Pull request by kridai**
# Fixes #

Added support to parse enum changes added in OAI spec for go as part of the PR: https://github.com/twilio/twilio-oai/pull/71

updated to spec file for related enum changes
### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] Run `make test-docker`
- [ ] Verify affected language:
    - [ ] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
    - [ ] Run `make test` in `twilio-go`
    - [ ] Create a pull request in `twilio-go`
    - [ ] Provide a link below to the pull request
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please create a GitHub Issue in this repository.

### CODE REVIEW

File: src/main/java/com/twilio/oai/AbstractTwilioGoGenerator.java [hunk 1]
```diff
@@ -113,6 +112,21 @@ public OperationsMap postProcessOperationsWithModels(final OperationsMap objs, L
         return objs;
     }
 
+    private Map<String, ModelsMap> filterOutEnumResults( Map<String, ModelsMap> results) {
+        HashMap <String, ModelsMap> resultMap = new HashMap<>();
+        results.forEach((key ,value) -> {
+            if (!key.contains("_enum_")) {
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/AbstractTwilioGoGenerator.java:118]: Is there a ticket in the backlog to revisit this?

File: src/main/java/com/twilio/oai/TwilioGoGenerator.java [hunk 2]
```diff
@@ -63,6 +63,44 @@ public ModelsMap postProcessModels(final ModelsMap objs) {
...
+            property._enum = (List<String>) property.items.allowableValues.get("values");
+            property.allowableValues = property.items.allowableValues;
+            property.dataType = "[]string";
+            property.isEnum = property.dataType == null;
+            property.isNullable = true;
+        } else if (property.dataType.contains("Enum")) {
+            String[] value = property.dataType.split("Enum");
+            property.datatypeWithEnum = value[value.length-1];
+            property.dataType = "string";
+            property.isEnum = property.dataType == null;;
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/TwilioGoGenerator.java:79]: This would always be false.

**Review by childish-sambino — APPROVED** [review]

File: src/main/java/com/twilio/oai/TwilioGoGenerator.java (hunk 2 above)
Comment (charan678) [reply src/main/java/com/twilio/oai/TwilioGoGenerator.java:79]: Yes, This should be property.isEnum = property.dataFormat == null;; Will fix it before merging

File: src/main/java/com/twilio/oai/AbstractTwilioGoGenerator.java (hunk 1 above)
Comment (charan678) [reply src/main/java/com/twilio/oai/AbstractTwilioGoGenerator.java:118]: Nope, We will create it tomorrow and we will also create other improvement tickets for preview and alternative approach to __enum__ name in OAI schema models

**Pull request merged**